kvbc/src/replica_resources.cpp

```diff
@@ -0,0 +1,59 @@
+// Copyright 2022 VMware, all rights reserved
+
+#include "replica_resources.h"
+
+using namespace concord::performance;
+
+void ReplicaResourceEntity::addMeasurement(const ISystemResourceEntity::measurement &m) {
+  if (m.type == ISystemResourceEntity::type::pruning_utilization) {
+    pruningUtilization_.addDuration({m.start, m.end});
+  }
+  if (m.type == ISystemResourceEntity::type::post_execution_utilization) {
+    postExecUtilization_.addDuration({m.start, m.end});
+  }
+  if (m.type == ISystemResourceEntity::type::pruning_avg_time_micro) {
+    ConcordAssertGE(m.end, m.start);
+    pruning_accumulated_time += (m.end - m.start);
+    ++pruned_blocks_;
+  }
+  if (m.type == ISystemResourceEntity::type::pruning_avg_time_micro) {
+    ConcordAssertGE(m.end, m.start);
+    pruning_accumulated_time += (m.end - m.start);
+    ++pruned_blocks_;
+  }
+  if (m.type == ISystemResourceEntity::type::consensus_throughput) {
+    numOfConsensus_ += m.count;
+  }
+  numOfExternalMsgs_ += IncomingMsgsStorageImp::num_of_external_msgs_;
+  ++numOfExternalMsgsCalls_;
+}
+
+uint64_t ReplicaResourceEntity::getMeasurement(const ISystemResourceEntity::type type) const {
```

Comment:
This switch should be replaced with polymorphism.

Reply:
In my opinion, using polymorphism, in this case, will introduce a lot of boilerplate code and complexity

Reply:
also, it makes the interface less vulnarable to changes, since adding a measurement type doesn't require changing the interface and all it's implementations.